lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -1773,6 +1765,16 @@ static bool IsDWARFImported(swift::ModuleDecl &module) {
                      });
 }
 
+/// Detect whether this is a proper Swift module.
+static bool IsSerializedAST(swift::ModuleDecl &module) {
+  return std::any_of(module.getFiles().begin(), module.getFiles().end(),
+                     [](swift::FileUnit *file_unit) {
+                       return (file_unit->getKind() ==
+                               swift::FileUnitKind::SerializedAST);
+                     });
```

Comment:
there's also `llvm::any_of` from STLExtras.h:
```suggestion
  return llvm::any_of(module.getFiles(),
                     [](swift::FileUnit *file_unit) {
                       return (file_unit->getKind() ==
                               swift::FileUnitKind::SerializedAST);
                     });
```